internal/dataplane/parser/ingressrules.go

```diff
@@ -290,6 +290,7 @@ func servicesAllUseTheSameKongAnnotations(
 	kongServiceName string,
 ) bool {
 	match := true
+	badAnnotations := map[string]interface{}{}
```

Comment:
Are we just recording the annotation keys? if so, it is better to use `map[string]struct{}` here.

Reply:
apimachinery provides a generic equivalent with fun set functions to boot, may as well use that.

Reply:
You're probably talking about 	"k8s.io/apimachinery/pkg/util/sets" right? We're already using that in a couple of places in  here